moveos/moveos-store/src/event_store/mod.rs

```diff
@@ -154,15 +156,42 @@ impl EventDBStore {
         let ids = if descending_order {
             let start = cursor.unwrap_or(last_seq + 1);
             let end = if start >= limit { start - limit } else { 0 };
+
+            let gaps = match end.checked_sub(start) {
+                None => return Err(anyhow::Error::msg("end value is overflow")),
+                Some(v) => v,
+            };
+            if gaps > MAX_RESULT_AMOUNT {
+                return Err(anyhow::Error::msg("end value is overflow"));
+            }
```

Comment:
Same as above

Reply:
This function is called by the `get_events_by_event_handle()` function in the `server/rooch_server.rs` file, but the code `(start..end).collect::<Vec<_>>()` is only executed internally within this function.

Reply:
Check the limit argument is enough.

Reply:
Fixed